scripts/dts/gen_defines.py

```diff
@@ -732,6 +695,64 @@ def write_vanilla_props(node):
         out_comment("(No generic property macros)")
 
 
+def string_macros(macro, val):
+    # Returns a dict of macros for a string 'val'.
+    # The 'macro' argument is the N_<node-id>_P_<prop-id>... part.
+
+    as_token = edtlib.str_as_token(val)
+    return {
+        # DT_N_<node-id>_P_<prop-id>_IDX_<i>_STRING_UNQUOTED
+        f"{macro}_STRING_UNQUOTED": val,
+        # DT_N_<node-id>_P_<prop-id>_IDX_<i>_STRING_TOKEN
+        f"{macro}_STRING_TOKEN": as_token,
+        # DT_N_<node-id>_P_<prop-id>_IDX_<i>_STRING_UPPER_TOKEN
+        f"{macro}_STRING_UPPER_TOKEN": as_token.upper()}
+
+
+def enum_macros(prop, macro):
+    # Returns a dict of macros for property 'prop' with a defined enum in their dt-binding.
+    # The 'macro' argument is the N_<node-id>_P_<prop-id> part.
+
+    spec = prop.spec
+    # DT_N_<node-id>_P_<prop-id>_IDX_<i>_ENUM_IDX
+    ret = {f"{macro}_IDX_{i}_ENUM_IDX": index for i, index in enumerate(prop.enum_indices)}
+    val = prop.val_as_token if spec.enum_tokenizable else (prop.val if isinstance(prop.val, list) else [prop.val])
+
+    for i, subval in enumerate(val):
+        # DT_N_<node-id>_P_<prop-id>_IDX_<i>_ENUM_VAL_<val>_EXISTS 1
+        ret[macro + f"_IDX_{i}_ENUM_VAL_{subval}_EXISTS"] = 1
+        if not spec.enum_tokenizable:
+            continue
+
+        # DT_N_<node-id>_P_<prop-id>_IDX_<i>_ENUM_TOKEN
+        ret[macro + f"_IDX_{i}_ENUM_TOKEN"] = subval
+        if spec.enum_upper_tokenizable:
+            # DT_N_<node-id>_P_<prop-id>_IDX_<i>_ENUM_UPPER_TOKEN
+            ret[macro + f"_IDX_{i}_ENUM_UPPER_TOKEN"] = subval.upper()
```

Comment:
why do they even exist, if there are already string tokens? Couldn't these few lines just be removed?

Reply:
Well observed. These are already deprecated, see devicetree.h, line 50ff. Maybe a good next PR for you to remove those everywhere? They've been deprecated for 3 years now, it seems. :-D

Thanks for factoring this code into methods, btw. Much more readable now.